doc/api/modules.md

```diff
@@ -377,9 +377,9 @@ If the exact filename is not found, then Node.js will attempt to load the
 required filename with the added extensions: `.js`, `.json`, and finally
 `.node`.
 
-`.js` files are interpreted as JavaScript text files, and `.json` files are
-parsed as JSON text files. `.node` files are interpreted as compiled addon
-modules loaded with `process.dlopen()`.
+`.json` files are parsed as JSON text files, `.node` files are interpreted as
+compiled addon modules loaded with `process.dlopen()`. Files using an unknown
+extension (or no extension at all) are parsed as JavaScript text files.
```

Comment:
“JavaScript text files” doesn’t tell me whether it’s as a Script/CJS, or as ESM, since JavaScript text files have two parse goals. (i know the answer - that it’s CJS - but it’d be good if this sentence was unambiguous)

Reply:
I think the wording here is carefully chosen so it applies to both CJS and ESM. Any non-`.json`-nor-`.node` extension is treated as JS text file, which may be either a module (in which case the require call will throw) or a script. I'm not a fan of ambiguity myself, but I don't think here is the place to list the rules to determine if a JS text file is parsed as script or module. wdyt?

Reply:
I think if ambiguity is the intention, your PR as-is indeed captures the ambiguity correctly :-)

Reply:
The CommonJS loader _does_ parse JS as CommonJS scripts always though, so I think @ljharb's seeking a clarification does make sense - there isn't ambiguity in play.

Reply:
Calling `require` on an ES module doesn't make it CJS, it throws a `ERR_REQUIRE_ESM`. Replacing `JavaScript text files` with `CommonJS modules` would not be an improvement imho. Do you have a suggestion that removes the ambiguity?

Reply:
Yes you're right, `.mjs` and `.js` in `"type": "module"` are excluded by this error, which does form an ambiguity of sorts.

Perhaps then explain exactly that, and that will solidify the ESM distinction:

> `.mjs` or `.js` files within a `"type": "module"` package boundary throw an `ERR_REQUIRE_ESM` error. All other files are parsed as CommonJS scripts.

Reply:
What if we link to the ad-hoc section in `packages.md`?

Reply:
Except for json and .node and wasm files.

Reply:
`require('./file.wasm')` loads `file.wasm` as CJS:

```sh
mkdir repro
cd repro
echo 'console.log(this !== undefined)' > file.wasm
echo 'require("./file.wasm")' | node
```